mirrord-layer/src/pod_api.rs

```diff
@@ -57,7 +58,7 @@ pub async fn create_agent(
     agent_namespace: &str,
     log_level: String,
     agent_image: String,
-) -> Portforwarder {
+) -> Result<Portforwarder> {
```

Comment:
I would suggest returning just `PortForwarder` here, and change `unwrap`s inside this function to `expect` with a nice message giving additional context.

If this function fails we're not handling a potential recovery yet, so it's better to just crash on failures.